blockchain/state_migration.go

```diff
@@ -371,3 +372,138 @@ func (bc *BlockChain) StopStateMigration() error {
 func (bc *BlockChain) StatusStateMigration() (bool, uint64, int, int, float64) {
 	return bc.db.InMigration(), bc.db.MigrationBlockNumber(), bc.committedCnt, bc.pendingCnt, bc.progress
 }
+
+func (bc *BlockChain) concurrentIterateTrie(root common.Hash, db state.Database, resultCh chan common.Hash, finishCh chan error) (resultErr error) {
```

Comment:
Why don't you implement this function in `state` package?

Reply:
Right. I will consider to move iteration code to `state` package.